gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfig.java

```diff
@@ -54,27 +55,49 @@ public interface ObservabilityConfig {
   /**
    * POJO for representing a filter used in configuration.
    */
+  @ThreadSafe
   class LogFilter {
-    /** Pattern indicating which service/method to log. */
-    public final String pattern;
+    /** Pattern for service/method filter. */
+    public final List<String> pattern;
 
-    /** Number of bytes of each header to log. */
+    /** Boolean to indicate all services and methods. */
+    public final Boolean matchAll;
+
+    /** Set of services. */
+    public final Set<String> services;
+
+    /* Set of fullMethodNames. */
+    public final Set<String> methods;
+
+    /** Number of bytes of header to log. */
     public final Integer headerBytes;
 
-    /** Number of bytes of each header to log. */
+    /** Number of bytes of message to log. */
     public final Integer messageBytes;
 
+    /** Boolean to indicate if services and methods matching pattern needs to be excluded. */
+    public final Boolean excludePattern;
```

Comment:
I wonder why you have to use boxed types like `Boolean` and `Integer` . Can you use primitive types instead since they all seem to have default values - so you don't really need `null` to represent an unspecified value.

Reply:
The fields added prior to this were boxed types, I followed the same pattern. But I agree with your reasoning, converted to unboxed types.